source/MaterialXGenShader/ShaderNode.h

```diff
@@ -47,6 +47,8 @@ class ShaderPort : public std::enable_shared_from_this<ShaderPort>
 
     ShaderPort(ShaderNode* node, const TypeDesc* type, const string& name, ValuePtr value = nullptr);
 
+    virtual ~ShaderPort() = default;
```

Comment:
If virtual methods are not used I would remove this, to avoid the extra vtable pointer being stored on each instance. It's an extra 8 bytes of data per port in the scene, so not a big deal until we have hundred thousand nodes with millions of ports in total. But if it's easy to remove we should not waste the data :)

Reply:
Didn't think about the vtable. I'll remove it as it's not needed.